File: 2024/src/01/day01b.m
% Open the file and read it line by line
fileID = fopen('input/day01.txt', 'r');

lefties = [];
righties = [];

% Read input
while ~feof(fileID)
    line = fgetl(fileID);
    
    parts = strsplit(strtrim(line), ' ');
    lefties = [lefties, str2double(parts{1})];
    righties = [righties, str2double(parts{2})];
end

% Process
result = 0
for l = lefties
    result = result +  sum(righties == l)*l;
end
disp(result)
fclose(fileID);

File: 2024/src/01/day01b.cc
#include <bits/stdc++.h>
using namespace std;

int main()
{
    vector<int> lefties;
    vector<int> righties;

    // Read input
    ifstream file{"../input/day01.in"};
    string line;
    while (getline(file, line))
    {
        // Parse via stringstream
        // a-b identifier: word number
        int a, b;
        stringstream ss(line);
        ss >> a >> b;
        lefties.push_back(a);
        righties.push_back(b);
    }

    // Process input
    long long result = accumulate(lefties.begin(), lefties.end(), 0, [&](long long sum, int l)
                                  { return sum + count(righties.begin(), righties.end(), l) * l; });
    cout << result;
    return 0;
}